pyspecdata/core.py

```diff
@@ -1623,43 +1618,32 @@ def concat(datalist, dimname, chop=False):
         )
     other_info_out = datalist[0].other_info
     if dimname in datalist[0].dimlabels:
-        print(
-            "The dimname you specified exists already in one of the datasets so I am assuming it also exists in the other"
+        assert all(
+            [dimname in datalist[j].dimlabels for j in range(len(datalist))]
+        ), (
+            "the dimension '"
+            + dimname
+            + "' exists in the first datset, but not the second!"
         )
         # {{{ make list of the data axes and dimname axes for datasets
         ndarrays = []  # will contain numpy array of data for each dataset
-        conc_axis = [] # will contain numpy array of the dimname axis for each dataset
+        conc_axis = (
+            []
+        )  # will contain numpy array of the dimname axis for each dataset
         for j in range(len(datalist)):
             ndarrays.append(datalist[j].data)
             conc_axis.append(datalist[j][dimname])
+        # PR: much more compact to just do this with a list comprehension in place, below
         # }}}
         # {{{ concatenate the data ndarrays and dimname axis ndarrays
-        newdata = np.concatenate(
+        concated_data = np.concatenate(
```

Comment:
I changed the names here -- not just for yours but for the stuff below, which I hadn't read in a while